### PR TITLE
feat: add api to delete a community channel

### PR DESCRIPTION
**Pull request by jrainville**
Adds the API and manager function to delete a community channel

### CODE REVIEW

**Comment by cammellos** [comment]
@jrainville there's a missing functionality I believe,
we need to notify the user that chats have been removed (in status-go).

For example when we leave a community, we add the removed chats in https://github.com/status-im/status-go/blob/8ac498a5fec39a9e4cfc3df52fe79a05843a7c68/protocol/messenger_communities.go#L367 

Similarly, we want to notify the user that a chat has been removed if the user has joined the chat,
probably the best place to do that is here:
https://github.com/status-im/status-go/blob/8ac498a5fec39a9e4cfc3df52fe79a05843a7c68/protocol/messenger_communities.go#L744

What do you think?

**Comment by jrainville** [comment]
@cammellos I just tried and the channel gets deleted in the other app as well. I know I put the `AddRemovedChat` already. Maybe the other line is not needed?

File: protocol/communities/manager.go [hunk 1]
```diff
@@ -334,6 +334,30 @@ func (m *Manager) EditChat(communityID types.HexBytes, chatID string, chat *prot
...
 
+func (m *Manager) DeleteChat(communityID types.HexBytes, chatID string) (*Community, *protobuf.CommunityDescription, error) {
+	community, err := m.GetByID(communityID)
+	if err != nil {
+		return nil, nil, err
+	}
+	if community == nil {
+		return nil, nil, ErrOrgNotFound
+	}
+	description, err := community.DeleteChat(chatID)
```
Comment (richard-ramos) [inline protocol/communities/manager.go:345]:
```
// Remove communityID prefix from chatID if exists
if strings.HasPrefix(chatID, communityID.String()) {
	chatID = strings.TrimPrefix(chatID, communityID.String())
}
```

Comment (richard-ramos) [inline protocol/communities/manager.go:357]:
```suggestion

        // Remove communityID prefix from chatID if exists
	if strings.HasPrefix(chatID, communityID.String()) {
		chatID = strings.TrimPrefix(chatID, communityID.String())
	}

	description, err := community.DeleteChat(chatID)
```

**Review by richard-ramos — APPROVED** [review]

**Comment by iurimatias** [comment]
tested in https://github.com/status-im/status-desktop/pull/2757

**Pull request merged**